src/rpdk/core/contract/resource_client.py

```diff
@@ -334,6 +334,29 @@ def generate_invalid_update_example(self, create_model):
         example = override_properties(self.invalid_strategy.example(), overrides)
         return {**create_model, **example}
 
+    def compare(self, inputs, outputs):
```

Comment:
*thought: i think we should start using annotations (like `def compare(self, inputs: dict, outputs: dict) -> None:`) or something similar so you know what are the possible inputs.